ui/app/rif/pages/index.js

```diff
@@ -27,34 +31,87 @@ const pageNames = {
   },
 }
 
-function buildScreen (screenComponent, currentView) {
-  return h(screenComponent, {key: currentView.screenName})
+function buildScreen (screenComponent, context, dispatch) {
```

Comment:
This is great. 

We should think about generalize this more. 

For example, the following code: 

```
if (context.params.showDomainsSearch) {
    searchDomains = (<SearchDomains/>);
  }
```

relates to RNS stuff, but we can make this component more abstract, treating that as a header or generic searchbar. Imagine that in a future we have other screens that have the same structure but are not RNS related.

Reply:
Yeah maybe we should create a new component called header or something like that and then inside put the search bar probably, we should make that more generic in the future, i agree, we can refactor that later if we need to.